cluster.go

```diff
@@ -908,9 +922,17 @@ func (c *ClusterClient) reaper(idleCheckFrequency time.Duration) {
 	}
 }
 
+func (c *ClusterClient) WrapPipelineExecer(fn func(func([]Cmder) error) func([]Cmder) error) {
+	c.wrapPipelineExecer = fn
+}
+
 func (c *ClusterClient) Pipeline() Pipeliner {
+	pipelineExec := c.pipelineExec
+	if c.wrapPipelineExecer != nil {
+		pipelineExec = c.wrapPipelineExecer(pipelineExec)
+	}
```

Comment:
Please change the code to follow existing pattern:

```
	if c.process != nil {
		return c.process(cmd)
	}
	return c.defaultProcess(cmd)
```

Reply:
Pipeline() is different from Process(), Pipeline() return a pipeline object which can be execed at later time, Process() exec command immediately. Did you mean something like
```
	if c.wrapPipelineProcess != nil {
		pipe := Pipeline{
			exec: c.wrapPipelineProcess(c.pipelineExec),
		}
		pipe.setProcessor(pipe.Process)
		return &pipe
	}
	pipe := Pipeline{
		exec: c.pipelineExec,
	}
	pipe.setProcessor(pipe.Process)
	return &pipe
```
or something like
```
	pipelineExec := func() func([]Cmder) error {
		if c.wrapPipelineProcess != nil {
			return c.wrapPipelineProcess(c.pipelineExec)
		}
		return c.pipelineExec
	}()

	pipe := Pipeline{
		exec: pipelineExec,
	}
	pipe.setProcessor(pipe.Process)
	return &pipe
```